### PR TITLE
schema included properties

### PR DESCRIPTION
**Pull request by ig-robstoffers**
Adding property to let users include specific properties in the schema.

### CODE REVIEW

File: ODataVirtualDataSourceDataProviderWorker.ts [hunk 1]
```diff
@@ -468,6 +471,25 @@ export class ODataVirtualDataSourceDataProviderWorker extends AsyncVirtualDataSo
 		let success_: (arg1: string) => void = (res: string) => {
 			let sp: ODataSchemaProvider = new ODataSchemaProvider(res);
 			let schema = sp.getODataDataSourceSchema(this._entitySet);
+			if (this._schemaIncludedProperties != null) {
```
Comment (gmurray81) [inline ODataVirtualDataSourceDataProviderWorker.ts:479]: Should use a hashset here

**Review by gmurray81 — CHANGES_REQUESTED** [review]

File: ODataVirtualDataSourceDataProviderWorker.ts (hunk 1 above)
Comment (ig-robstoffers) [reply ODataVirtualDataSourceDataProviderWorker.ts:479]: I've adjusted things to use a hash set in order to check if the schema props should be included.

File: ODataVirtualDataSourceDataProviderWorker.ts [hunk 2]
```diff
@@ -475,14 +480,7 @@ export class ODataVirtualDataSourceDataProviderWorker extends AsyncVirtualDataSo
...
 				for (let i = 0; i < schema.propertyNames.length; i++) {
-					let found = false;
-					for (let j = 0; j < this._schemaIncludedProperties.length; j++) {
-						if (this._schemaIncludedProperties[j] == schema.propertyNames[i]) {
-							found = true;
-							break;
-						}
-					}
-					if (!found) {
+					if (!this._schemaIncludedProperties.has(schema.propertyNames[i])) {
```
Comment (gmurray81) [inline ODataVirtualDataSourceDataProviderWorker.ts:483]: You need to include everything If the included set is null though

**Review by gmurray81 — CHANGES_REQUESTED** [review]

File: ODataVirtualDataSourceDataProviderWorker.ts (hunk 2 above)
Comment (ig-robstoffers) [reply ODataVirtualDataSourceDataProviderWorker.ts:483]: The code should do that already if the _schemaIncludedProperties set is null. There's a null check a few lines above this. If there is a null set then it does not alter the schema so all the properties are included. It only filters out the props if there is a valid set.

**Review by gmurray81 — APPROVED** [review]

File: RestVirtualDataSourceDataProviderWorker.ts [hunk 3]
```diff
@@ -625,6 +633,18 @@ export class RestVirtualDataSourceDataProviderWorker extends AsyncVirtualDataSou
...
+					let propertyNames: any[] = [];
+                	let propertyTypes: any[] = [];
+					for (let i = 0; i < schema.propertyNames.length; i++) {
+						if (!this._schemaIncludedProperties.has(schema.propertyNames[i])) {
+							continue;
+						}
+						propertyNames.push(schema.propertyNames[i]);
+						propertyTypes.push(schema.propertyTypes[i]);
+					}
+					schema = new DefaultDataSourceSchema(propertyNames, propertyTypes, schema.primaryKey, schema.propertyDataIntents);
```
Comment (IMinchev64) [inline RestVirtualDataSourceDataProviderWorker.ts:646]: I know the PR has already been merged but isn't the constructor here missing a 5th argument subSchemas?

**Pull request merged**